test/check-subscriptions

```diff
@@ -0,0 +1,232 @@
+#
+# add an activation key to a pool:
+# curl --insecure --request POST --user admin:admin https://$IP:8443/candlepin/activation_keys/ff80808155ca50b10155ca50cd280010/pools/ff80808155ca50b10155ca51f04607e5
+# register with: activation key "awesome_os_pool" and org "admin"
+# or: subscription-manager register --activationkey awesome_os_pool --org admin --serverurl https://$IP:8443/candlepin
+#
+# in order to get the right ids for the activation key and pool, see ACTIVATION_KEY_COMMAND and POOL_COMMAND
+
+WAIT_SCRIPT = """
+set -ex
+for x in $(seq 1 200); do
+    if curl --insecure -s https://%(addr)s:8443/candlepin; then
+        break
+    else
+        sleep 1
+    fi
+done
+"""
+
+ACTIVATION_KEY_COMMAND = """curl -s --insecure --request GET --user admin:admin https://localhost:8443/candlepin/activation_keys | python -c "
+import sys
+import json
+
+data = json.loads(sys.stdin.read())
+print [e['id'] for e in data if e['name'] == 'awesome_os_pool' and e['owner']['displayName'] == 'Admin Owner'][0]
+"
+"""
+
+POOL_COMMAND = """curl -s --insecure --request GET --user admin:admin https://localhost:8443/candlepin/pools | python -c "
+import sys
+import json
+
+data = json.loads(sys.stdin.read())
+print [ e['id'] for e in data if e['owner']['key'] == 'admin' and e['contractNumber'] == '0' and [p for p in e['providedProducts'] if p['productId'] == '88888'] ][0]
+"
+"""
+
+class TestSubscriptions(MachineCase):
+    provision = {
+        "0": {"address": "10.111.113.1/20"},
+        "candlepin": {"image": "candlepin", "address": "10.111.113.5/20"}
+    }
+
+    def setUp(self):
+        MachineCase.setUp(self)
+        self.candlepin = self.machines['candlepin']
+
+        # wait for candlepin to be active and verify
+        self.candlepin.execute("systemctl start tomcat")
+
+        # download product info from the candlepin machine
+        product_file_one = os.path.join(self.tmpdir, "6050.pem")
+        product_file_two = os.path.join(self.tmpdir, "88888.pem")
+        self.candlepin.download("/home/admin/candlepin/generated_certs/6050.pem", product_file_one)
+        self.candlepin.download("/home/admin/candlepin/generated_certs/88888.pem", product_file_two)
+
+        # upload product info to the test machine
+        m = self.machine
+        m.upload([product_file_one, product_file_two], "/etc/pki/product")
+
+        # make sure that rhsm skips certificate checks for the server
+        m.execute("sed -i -e 's/insecure = 0/insecure = 1/g' /etc/rhsm/rhsm.conf")
+
+        # Apply some extra cleanups on rhel-atomic.  These cleanups
+        # are necessary because all changes to /etc after a "atomic
+        # host upgrade" and before the next boot are lost.
+        if m.image == "rhel-atomic":
+            m.execute("rm -f /etc/pki/consumer/* /etc/pki/entitlement/*")
+
+        # Wait for the web service to be accessible
+        args = {"addr": "10.111.113.5"}
+        m.execute(script=WAIT_SCRIPT % args)
+
+    def testRegister(self):
```

Comment:
The overhead to setup a test is significant, unfortunately, so we tend to bunch a couple of tests together into a single run just for performances sake... But I see your point, of course.  I'll split them up.

Reply:
> The overhead to setup a test is significant, unfortunately, so we tend to bunch a couple of tests together into a single run just for performances sake... But I see your point, of course. I'll split them up.

You needn't to use `setUp()` for starting tomcat. You can also use `setUpClass()` for this purpose. https://docs.python.org/3/library/unittest.html#unittest.TestCase.setUpClass

Reply:
> You can also use setUpClass() for this purpose.

Ohh, nice!  I'll use that.

Reply:
> > You can also use setUpClass() for this purpose.
> 
> Ohh, nice! I'll use that.

That didn't work out as expected.  The serUpClass method can't start any machines since that needs an instance.  (I guess we can try adding support for "infrastructure" machines that are meant to be shared between tests.)

Reply:
OK.